### PR TITLE
CASMPET-6004 CASMPET-6018 CASMPET-6019 Encryption enhancements 1.3

### PR DESCRIPTION
**Pull request by mitchty**
# Description

Updates to encryption.sh script based on testing feedback and issues encountered.

This adds a --status switch to encryption.sh to get detailed status of where encryption sits across the cluster. And a --restart switch which will force an encryption rewrite and restart all pods to enable rewrites of data.

This also adds the ability for encryption.sh to read from etcd directly to determine if/when a user is requesting a configuration that would egregiously break kubernetes. This is the primary reason for the pr most of the rest is debugging related.

While here moved all shellspec code to scripts dir as per Dennis.

# Checklist Before Merging

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [ ] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Comment by mitchty** [comment]
Will fix the conflicts thought I fixed that already.

**Review by bklei — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

File: operations/kubernetes/encryption/README.md [hunk 1]
```diff
@@ -136,6 +147,10 @@ Encryption status is obtained through the `--status` switch of the `encryption.s
 
     ```text
     k8s encryption status
+    changed: 1970-01-01 00:20:00+0000
+    ncn-m001: aescbc-46d5bd8c2001d07ded05687fe51b517033dc609e69fe4dddaa6e05656cf6e907 identity
+    ncn-m002: aescbc-46d5bd8c2001d07ded05687fe51b517033dc609e69fe4dddaa6e05656cf6e907 identity
+    ncn-m003: aescbc-46d5bd8c2001d07ded05687fe51b517033dc609e69fe4dddaa6e05656cf6e907 identity
     current: aescbc-46d5bd8c2001d07ded05687fe51b517033dc609e69fe4dddaa6e05656cf6e907
     goal: aescbc-46d5bd8c2001d07ded05687fe51b517033dc609e69fe4dddaa6e05656cf6e907
     etcd: aescbc-46d5bd8c2001d07ded05687fe51b517033dc609e69fe4dddaa6e05656cf6e907
```
Comment (rgross727) [inline operations/kubernetes/encryption/README.md:156]: Could add at the end of "This indicates that all secret data in etcd is now encrypted with this key provider's name."
Note the return code for status is zero in this case.

File: operations/kubernetes/encryption/README.md [hunk 2]
```diff
@@ -31,6 +31,10 @@ For further information, refer to the [official Kubernetes documentation](https:
 
 ## Enabling encryption
 
+When enabling encryption it is important to ensure all 3 nodes are enabled in short order. However that does not mean all control plane nodes should run the script in parallel.
```
Comment (rgross727) [inline operations/kubernetes/encryption/README.md:34]: Some where in here, could add a paragraph that says:
Before encryption is enabled, it is recommend that a Bare-Metal etcd backup be saved (etcd backups are deleted after 24 hours) and that the health of the ectd cluster be verified.
See [Create_a_Manual_Backup_of_a_Healthy_Bare-Metal_etcd_Cluster.md](https://github.com/Cray-HPE/docs-csm/blob/70dc14a21024f4dd20f9416fdd5010c17a0b019b/operations/kubernetes/Create_a_Manual_Backup_of_a_Healthy_Bare-Metal_etcd_Cluster.md)
When encryption is enabled a Bare-Metal etcd cluster can not be restored from a backup taken before encryption is enabled. Such a backup can be used to restore etcd in the event that encryption is later disabled.

Comment (mitchty) [reply operations/kubernetes/encryption/README.md:38]: Pushed up a commit with most of this in place. Let me know if you think that addresses things.

**Review by rgross727 — APPROVED** [review]

**Comment by mitchty** [comment]
Should be set now.

**Pull request merged**